### PR TITLE
Feat/edit event

### PR DESCRIPTION
**Pull request by GagahGK**
| image | image | <br>image |
| ---- | ---- | ---- |
| My Event | Event Detail | Edit Event |

This PR add the following features :

- Ability to Edit Event https://github.com/inlivedev/inlive-room/pull/141/commits/e77ddc444ebad333dfe8ffa8a4f73905a51b69ea
- Ability to Delete Event  https://github.com/inlivedev/inlive-room/pull/141/commits/e40556dfafbd372515c5c8791d36dd69e714d98f https://github.com/inlivedev/inlive-room/pull/141/commits/9fa1fc8b9c05a3099bf4995ec24a9aee8e794528
- Update react-datepicker to fix crash on Firefox https://github.com/inlivedev/inlive-room/pull/141/commits/f650845337d9af7cf2df2fc7b2aa5c53f80b1700
- Added spinner to the button in create, edit, and delete operation
- Ability to navigate to event page using id
- Updated view in the Detail Event page https://github.com/inlivedev/inlive-room/pull/141/commits/9fa1fc8b9c05a3099bf4995ec24a9aee8e794528
	- Event Creator now can see the status of the event
	- Changed Action Button on Detail Event Page, event creator now can edit, delete, and share their event quickly from the event page 

This PR also fixes the following important changes :

- Updated FK references from `Event` table to Event_to_participant table, now when event is deleted it will perform cascade deletion the the Event_to_participant table https://github.com/inlivedev/inlive-room/pull/141/commits/e4cb70260276f8fe638b2972c74a077bb46bf94a
- a fix to the GetListEvent endpoint where the pagination didn't work properly https://github.com/inlivedev/inlive-room/pull/141/commits/fb81ca4c42a294fd384a12a80c0677b16bbb0824
- fixed datetime related bug https://github.com/inlivedev/inlive-room/pull/141/commits/f8ccc5698915e2cda4ed9cfad2e0414b0c85375f
- fix to the Update Event endpoint where it will remove the image when the it's updated without changing the image https://github.com/inlivedev/inlive-room/pull/141/commits/fc137599cd368dbc417336a5effdc15ccf61cbea
- The event image on event detail page was hardcoded now uses value from `thumbnailUrl` from database https://github.com/inlivedev/inlive-room/pull/141/commits/8f69b468c285c8554eaa97255bc397b0bc632000
- move the server.ts from the `app` folder so during development the event image will be saved to `/storage/` not `/app/storage`

### CODE REVIEW

File: app/_features/event/components/event-detail.tsx [hunk 1]
```diff
@@ -159,3 +155,124 @@ export default function EventDetail({
...
+              />
+              <span>Editing...</span>
+            </div>
+          ) : (
+            <div className="flex items-center gap-unit-2 align-middle">
+              <EditIcon height={20} width={20} />
+              <span>Edit Event</span>
+            </div>
+          )}
+        </Button>
```
Comment (faiq-naufal) [inline app/_features/event/components/event-detail.tsx:214]: Similar to [this comment](https://github.com/inlivedev/inlive-room/pull/142#discussion_r1478094200). This edit button is just a navigation link right? Please use `<Link>` nextjs component with the `<Button>` instead.

**Review by faiq-naufal — CHANGES_REQUESTED** [review]
Please check the comments. Thank you

**Comment by faiq-naufal** [comment]
@GagahGK  Between edit event button vs copy link button. Which one is the primary CTA button? I think edit event button is the one that needs to be emphasized so the user can immediately guided when some data needed to be updated.
302185264-141fbe08-4502-469a-8423-b4f49d8a010d

**Review by faiq-naufal — CHANGES_REQUESTED** [review]

**Comment by GagahGK** [comment]
> @GagahGK Between edit event button vs copy link button. Which one is the primary CTA button? I think edit event button is the one that needs to be emphasized so the user can immediately guided when some data needed to be updated. 302185264-141fbe08-4502-469a-8423-b4f49d8a010d

I think Copy Link should be the primary CTA, since compared to Editing Event the Copy Link button will be used more frequently, User will perform editing to the event only if there's something needs to be revised which don't happen. but copying link to share the event will be done more frequently instead

**Comment by GagahGK** [comment]
@faiq-naufal, i've revised to code as you requested : 

1. Card using Href : https://github.com/inlivedev/inlive-room/pull/141/commits/e580e2257daab4c36952f2fc0262a91fc277d346
2. `<Code>` tag now uses `<div>` instead : https://github.com/inlivedev/inlive-room/pull/141/commits/37340c6f77834d1046dd6d28f485125dfee7541c
3.  notFound() component issue : https://github.com/inlivedev/inlive-room/pull/141/commits/6d5798bddf94ddf7898a33b0e8724795e47b40d5
4.  about the default Local Storage Path : https://github.com/inlivedev/inlive-room/pull/141/commits/41d7588f6f2d18f4542adf055cbef8776e4a1452
5. oneliner code for getting cookie : https://github.com/inlivedev/inlive-room/pull/141/commits/d453c45df622bf5f7ca9da3f7cf9557aa85c21c3
6. remove useEffect for settign existing event : https://github.com/inlivedev/inlive-room/pull/141/commits/61f62c800639c83eeccdf3ff504b4e8e9f56188b https://github.com/inlivedev/inlive-room/pull/141/commits/8d3be5be6e517933e8ed04e16bc49fff23c0b7db

**Comment by faiq-naufal** [comment]
I see the default local storage path has changed into `./storage`. There are some places where you haven't changed the code. I fixed it here [2e2d4d8](https://github.com/inlivedev/inlive-room/pull/141/commits/2e2d4d89902b0d55cb1016c4c8847acb1c01f0a1)

**Comment by faiq-naufal** [comment]
> I think Copy Link should be the primary CTA, since compared to Editing Event the Copy Link button will be used more frequently, User will perform editing to the event only if there's something needs to be revised which don't happen. but copying link to share the event will be done more frequently instead

@GagahGK  I see, I think we also need insight from mas Ashar about this. But for now, I think it's fine to make the copy link primary cta.

**Review by faiq-naufal — APPROVED** [review]
LGTM

**Pull request merged**